src/improvements/tasks/MultisigTask.sol

```diff
@@ -826,16 +838,71 @@ abstract contract MultisigTask is Test, Script, StateOverrideManager {
         );
     }
 
-    /// @notice get the multicall address for the given safe
-    /// it will be the regular multicall address for parent as well as child multisigs
-    /// @param safe The address of the safe
-    /// @return The address of the multicall
-    /// @dev override to return required multicall address
+    /// @notice Get the multicall address for the given safe.
+    /// Override to return required multicall address.
     function _getMulticallAddress(address safe) internal view virtual returns (address) {
+        // Some child contracts may override this function and return a different multicall address
+        // based on the safe address (e.g. whether it's the parent or child multisig).
         require(safe != address(0), "Safe address cannot be zero address");
         return multicallTarget;
     }
 
+    /// @notice To show the full transaction trace in Tenderly, we build custom calldata
+    /// that shows both the child multisig approving the hash, as well as the parent multisig
+    /// executing the task. This is only used when simulating a nested multisig.
+    function getNestedSimulationMulticall3Calldata(Action[] memory actions, address childMultisig)
+        internal
+        view
+        virtual
+        returns (bytes memory data)
+    {
+        Call3Value[] memory calls = new Call3Value[](2);
+
+        (bytes memory approveHashCallData,,,) = getApproveTransactionInfo(actions, childMultisig);
+        bytes memory approveHashExec = _execTransationCalldata(
+            childMultisig,
+            approveHashCallData,
+            Signatures.genPrevalidatedSignature(MULTICALL3_ADDRESS),
+            MULTICALL3_ADDRESS
+        );
+        calls[0] = Call3Value({target: childMultisig, allowFailure: false, value: 0, callData: approveHashExec});
+
+        bytes memory customExec = _execTransationCalldata(
+            parentMultisig,
+            getMulticall3Calldata(actions),
+            Signatures.genPrevalidatedSignature(childMultisig),
+            _getMulticallAddress(parentMultisig)
+        );
+        calls[1] = Call3Value({target: parentMultisig, allowFailure: false, value: 0, callData: customExec});
+
+        return abi.encodeWithSignature("aggregate3Value((address,bool,uint256,bytes)[])", calls);
+    }
+
+    /// @notice Helper function to get the approve transaction info.
+    function getApproveTransactionInfo(Action[] memory actions, address childMultisig)
+        internal
+        view
+        returns (bytes memory callData, bytes memory encodedTxData, bytes32 domainSeparator, bytes32 messageHash)
+    {
+        callData = generateApproveMulticallData(actions);
+        encodedTxData = getEncodedTransactionData(childMultisig, callData);
+        require(encodedTxData.length == 66, "MultisigTask: encodedTxData length is not 66 bytes.");
```

Comment:
This check seems like it should live in the `getEncodedTransactionData` method

Reply:
Implemented here: https://github.com/ethereum-optimism/superchain-ops/pull/771/commits/2979b3c74d89bba4ca8fb0c6774bb122400ecad3